### PR TITLE
MeasurmentPlot new params

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/zakodium-oss/analysis-ui-components/issues/144
closes : https://github.com/zakodium-oss/analysis-ui-components/issues/173

### CODE REVIEW

**Comment by lpatiny** [comment]
Here is the direct link: https://deploy-preview-166--analysis-ui-components.netlify.app/storybook/?path=/story/layout-measurementexplorer--control

Changing the variable (x / y) crash the system.

Could you also update the storybook to be able to activate / desactivate the different new options ?

**Comment by wadjih-bencheikh18** [comment]
> Changing the variable (x / y) crash the system.

I added an error that you have to enter valid variable

**Comment by lpatiny** [comment]
But do you know why the spectrum disappear ?

**Comment by wadjih-bencheikh18** [comment]
> But do you know why the spectrum disappear ?
> 

Yes i'm now fixing it

**Comment by lpatiny** [comment]
Playing around with zoom and change of variables crash

image

**Comment by wadjih-bencheikh18** [comment]
> Playing around with zoom and change of variables crash

bug fixed
there's one problem that  we cannot use the same variable

**Review by lpatiny — CHANGES_REQUESTED** [review]
There is a crash if you try to switch the axis

image

**Comment by wadjih-bencheikh18** [comment]
> There is a crash if you try to switch the axis
>

this problem is due to a contrainte in react-plot that force the user to use x as horizontal and y to vertical

i fixed this problem here but changing axisId 

```
  xAxis = `${xVariableName}-x`
  yAxis =`${yVariableName}-y`
```


> there's one problem that we cannot use the same variable

also with this i could fix same variable problem

File: src/components/MeasurementPlot.tsx [hunk 1]
```diff
@@ -72,6 +72,8 @@ function MeasurementComponent(props: MeasurementPlotProps) {
   } = props;
   const { title = '', data } = measurement;
 
+  const xAxis = useMemo(() => `${xVariableName}-x`, [xVariableName]);
+  const yAxis = useMemo(() => `${yVariableName}-y`, [yVariableName]);
```
Comment (targos) [inline src/components/MeasurementPlot.tsx:76]: Strings are always unique, you don't need to memo them.

Comment (wadjih-bencheikh18) [reply src/components/MeasurementPlot.tsx]: removed, thank you

**Pull request merged**